### DISCUSSION TITLE
How are tests that should fail indicated?

### DISCUSSION THREAD

**Discussion by stefandrissen** [Riverside-Software/pct#541] [Q&A] [answered]
While checking if release 229 contained changes for OpenEdge 13.0 I stumbled across the [build results for OE13](https://ci.rssw.eu/job/PCT/job/OE13/lastSuccessfulBuild/artifact/emailable-report-13.0-Linux.html). 

The summary shows all tests passed, but when looking at the details, for example of [ABLUnit test3](https://ci.rssw.eu/job/PCT/job/OE13/lastSuccessfulBuild/artifact/emailable-report-13.0-Linux.html#m3) - it shows 1 failure.

Looking at what I think are the [source files](https://github.com/Riverside-Software/pct/tree/main/tests/ABLUnit/test3) of this test, I cannot see what is indicating that this is an expected failure.

**Comment by gquerret** [comment, marked as answer]
Unit tests are executed in Java, here is what is executed: [link](https://github.com/Riverside-Software/pct/blob/main/src/test/com/phenix/pct/test/ABLUnitTest.java#L71)
By default, [ABLUnit](https://wiki.rssw.eu/pct/ABLUnit.md) tests do not fail the build in case of failure of the OE tests (attribute haltOnFailure set to false). So the Java unit test checks that the unit tests don't fail the build (expected, and that the output file contains 3 tests, 1 failure, 1 error, and so one success).

**Comment by stefandrissen** [comment]
Thanks! I was expecting the expected results to be closer to the tests themselves, ie in the test directories.
